### PR TITLE
Support for UQ and NICF with vectorized (SIMD) centered schemes

### PR DESCRIPTION
**Pull request by pcarruscag**
## Proposed Changes
This allows UQ and NICF features to be used with the vectorized centered schemes.
UQ can also be used with the vectorized Roe scheme, but not NICF as that would not be equivalent to the general Roe implementation.
With this the older implementations were removed as they are never used as boundary numerics.

## Related Work
#1022

## PR Checklist
- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [X] My contribution is commented and consistent with SU2 style.
- [X] I have added a test case that demonstrates my contribution, if necessary.
- [X] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: Common/include/linear_algebra/blas_structure.hpp [hunk 1]
```diff
@@ -143,6 +143,345 @@ class CBlasStructure {
...
+   * ISBN13: 978-3540075462,
+   * LC: QA193.M37
+   *
+   * \param[in,out] V: matrix that needs to be decomposed
+   * \param[in,out] d: holds eigenvalues
+   * \param[in,out] e: work vector
+   * \param[in] n: order of matrix V
+   */
+  template<class Mat, class Vec, class W>
+  static void tred2(Mat& V, Vec& d, W& e, int n) {
```
Comment (pcarruscag) [inline Common/include/linear_algebra/blas_structure.hpp:182]: @jayantmukho do you know any simple methods that would be stable enough on 3x3 problems to do what tred2 / tql2 do?
That is the only thing standing in the way of vectorizing the UQ stuff (too many if's in these routines).

**Comment by pcarruscag** [comment]
I was told last week that the numerics used for boundary terms pretty much never changed in 9 years.
I'll go ahead with removing the old centered compressible numerics and making the vectorized ones the default.

**Comment by maxaehle** [comment]
Valgrind indicated a memory leak associated to `CFVMFlowSolverBase::edgeNumerics`. Is it possible to `free` it in `~CFVMFlowSolverBase`? (I hope this is the correct place to discuss this.)

**Comment by pcarruscag** [comment]
Absolutely I'll have a look

**Comment by pcarruscag** [comment]
Fixed in #1159

**Pull request merged**